lib/sacastats_web/templates/character/weapons.html.heex

```diff
@@ -1,15 +1,193 @@
         </tr>
     </thead>
-    <tbody></tbody>
+    <tbody>
+        <%= for {weapon_id, weapon} <- @character["weapons"] do %>
+            <tr id={"weapon" <> (weapon_id |> Integer.to_string()) <> "Row"} class="">
+                <td class="weapon p-0">
+                    <%= if SacaStatsWeb.CharacterController.get_medal_code(SacaStatsWeb.CharacterController.get_total_values(Map.get(weapon, "weapon_kills")), 
+                            SacaStatsWeb.CharacterController.get_total_values(Map.get(weapon, "weapon_vehicle_kills"))) > 0 do%>
+                        <img src={"https://census.daybreakgames.com/files/ps2/images/static/" <> (SacaStatsWeb.CharacterController.get_medal_code(SacaStatsWeb.CharacterController.get_total_values(Map.get(weapon, "weapon_kills")), 
+                            SacaStatsWeb.CharacterController.get_total_values(Map.get(weapon, "weapon_vehicle_kills")))|> Integer.to_string()) <>".png"} alt="medal" class="medal float-left" width="30px"/>
+                    <% end %>
+                    <img src="https://census.daybreakgames.com/files/ps2/images/static/2021.png" alt={Map.get(weapon, "name", "N/A")} class="weapon mx-auto d-block" width="150px"/>
+                    <h5 class="weaponName align-text-bottom text-center mb-0"><%=Map.get(weapon, "name", "N/A")%></h5>
+                    <div id={"weapon" <> (weapon_id |> Integer.to_string()) <> "ProgressBar"} class="progress-bar progress-bar-striped progress-bar-animated aurax-progress" role="progressbar" 
+                        aria-valuenow={"#{SacaStatsWeb.CharacterController.get_aurax_percent(SacaStatsWeb.CharacterController.get_total_values(Map.get(weapon, "weapon_kills")), 
+                            SacaStatsWeb.CharacterController.get_total_values(Map.get(weapon, "weapon_vehicle_kills")))}"} 
+                        aria-valuemin="0" aria-valuemax="100" style="width: 0;">
+                        <%= SacaStatsWeb.CharacterController.get_aurax_percent(SacaStatsWeb.CharacterController.get_total_values(Map.get(weapon, "weapon_kills")), 
+                            SacaStatsWeb.CharacterController.get_total_values(Map.get(weapon, "weapon_vehicle_kills")))%>%
+                    </div>
+                </td>
+                <td id={"weapon" <> (weapon_id |> Integer.to_string()) <> "Kills"} class="number text-center total-kills">
+                    <%=SacaStatsWeb.CharacterController.get_total_values(Map.get(weapon, "weapon_kills")) + SacaStatsWeb.CharacterController.get_total_values(Map.get(weapon, "weapon_vehicle_kills"))%>
+                </td>
+                <td class="number text-center">
+                    <%=SacaStatsWeb.CharacterController.get_string_value_as_integer(weapon["weapon_kills"]["value_vs"]) + SacaStatsWeb.CharacterController.get_string_value_as_integer(weapon["weapon_vehicle_kills"]["value_vs"])%>
```

Comment:
```suggestion
                    <%=SacaStatsWeb.CharacterController.maybe_to_int(weapon["weapon_kills"]["value_vs"]) + SacaStatsWeb.CharacterController.maybe_to_int(weapon["weapon_vehicle_kills"]["value_vs"])%>
```
I won't change them all here, since it can be done a lot faster with search/replace in a proper code editor :sweat_smile: